lib/baby_squeel/join_dependency.rb

```diff
@@ -119,7 +119,7 @@ def build(relation, buckets)
 
         join_list = join_nodes + joins
 
-        alias_tracker = Associations::AliasTracker.create(relation.klass.connection, relation.table.name, join_list)
+        alias_tracker = relation.alias_tracker(join_list)
```

Comment:
```suggestion
        alias_tracker = Associations::AliasTracker.create(relation.klass.connection.pool, relation.table.name, join_list)
```
Not sure about this change but I found this open MR https://github.com/rzane/baby_squeel/pull/132/files and they did it like this

Reply:
But I am not sure if your solution is better

Reply:
hi @eglitobias 
I’m not sure which solution is better. Based on their changes in [this pull request](https://github.com/rzane/baby_squeel/pull/132/files), they haven’t updated the build.yml file, so their pipelines are still running on older ActiveRecord versions.

In contrast, I’m running the pipelines with the latest supported Rails version (7.2.2) and Ruby version (3.2.2). I’ve made updates according to the suggested merge request, but I believe we should run the pipelines with the changes we’re currently implementing to ensure compatibility and correctness.

Thanks!